ldapauthenticator/ldapauthenticator.py

```diff
@@ -142,6 +143,9 @@ def _validate_bind_dn_template(self, proposal):
 
         Set to an empty list or None to allow all users that have an LDAP account to log in,
         without performing any group membership checks.
+
+        When combined with `search_filter`, this strictly reduces the allowed users,
+        i.e. `search_filter` AND `allowed_groups` must both be satisfied.
```

Comment:
In practice, `search_filter` also constrains use of `allowed_users` and `admin_users` when its applied (when lookup_dn=True) and not just `allowed_groups` because mismatching the filter returns (None, None) from resolve_username.

Due to this, its not very allowing, its previous implementation is entirely blocking even if the docstring sais its allowing.

Reply:
`lookup_dn_search_filter` and `search_filter` are two different options. `search_filter` appears to be applied unconditionally as an allow step, is that wrong?

Reply:
Aaah thanks you are right, I got the `looked_dn` coupling wrong!